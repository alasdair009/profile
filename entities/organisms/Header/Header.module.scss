@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";

$headerHeight: dimensions.getSize(s48);

.root {
  background: rgba(colors.$colorGreyDark, 0.5);
  border-bottom: dimensions.getSize(s2) solid
    rgba(colors.$colorGreenGrass, 0.25);
  display: block;
  height: $headerHeight;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalDecorationMaxWidth);
  padding: dimensions.getSize(s8);
  position: sticky;
  width: 100%;
}

.inner {
  display: flex;
  height: 100%;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalContentMaxWidth);
  width: 100%;
}

.headerLogo {
  &:hover {
    filter: brightness(1.3);
  }
}

.headerLinks {
  align-items: center;
  display: flex;
  flex: 1;
  gap: dimensions.getSize(s16);
  height: 100%;
  justify-content: flex-end;
}

.headerLink {
  align-items: center;
  display: flex;
  flex-direction: row-reverse;
  height: 100%;
  justify-content: center;
  white-space: nowrap;

  &.hideOnNarrow {
    display: none;

    @media (map-get(dimensions.$device, xsmall)) {
      display: flex;
    }
  }
}

.notificationIcon {
  height: dimensions.getSize(s24);
  opacity: 0.3;
  width: dimensions.getSize(s24);

  &.isSubscribed {
    opacity: 1;
  }
}
