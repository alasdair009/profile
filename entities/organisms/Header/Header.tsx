import {
  Inner,
  Root,
  HeaderLogo,
  headerHeight,
  HeaderLinks,
  HeaderLink,
} from "./styles";
import { HTMLAttributes } from "react";
import amLogo from "../../assets/am.svg";
import { sizes } from "@/entities";
import Link from "next/link";

type HeaderProps = {} & HTMLAttributes<HTMLDivElement>;
export function Header({ children, ...rest }: HeaderProps) {
  return (
    <Root {...rest}>
      <Inner>
        <Link href="/">
          <HeaderLogo src={amLogo} alt="AM logo" height={sizes.s32.raw} />
        </Link>
        <HeaderLinks>
          <HeaderLink href="/">Home</HeaderLink>
          <HeaderLink href="/portfolio">Portfolio</HeaderLink>
          <HeaderLink href="/about-me">About Me</HeaderLink>
        </HeaderLinks>
      </Inner>
    </Root>
  );
}
