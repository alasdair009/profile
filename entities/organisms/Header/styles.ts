import styled from "styled-components";
import {rem, rgba} from "polished";
import {colors} from "../../design-tokens/colors";
import {globalContentMaxWidth, globalDecorationMaxWidth, sizes} from "../../design-tokens/dimensions";
import Image from "next/image";

export const headerHeight = sizes.s48.rem;

export const Root = styled.header`
  background: ${rgba(colors.greyDark, 0.5)};
  border-bottom: ${sizes.s2.rem} solid ${rgba(colors.greenGrass, 0.25)};
  display: block;
  height: ${headerHeight};
  margin: 0 auto;
  max-width: ${rem(globalDecorationMaxWidth)};
  padding: ${sizes.s8.rem};
  position: sticky;
  width: 100%;
`;

export const Inner = styled.nav`
  display: flex;
  margin: 0 auto;
  max-width: ${rem(globalContentMaxWidth)};
  width: 100%;
`;

export const HeaderLogo = styled(Image)`
  &:hover {
    filter: brightness(1.3);
  }
`;