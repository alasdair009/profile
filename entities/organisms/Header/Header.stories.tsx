import {Header} from "./Header";
import {Meta, StoryObj} from "@storybook/react";

const meta: Meta<typeof Header> = {
    component: Header,
}
export default meta;

export const Default: StoryObj<typeof Header> = {};