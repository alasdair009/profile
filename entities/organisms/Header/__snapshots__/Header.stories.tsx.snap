// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/Header Default smoke-test 1`] = `
<header class="sc-dKsqdn idjsjM">
  <nav class="sc-eowDPD cdvzBf">
    <a href="/">
      <img alt="AM logo"
           loading="lazy"
           width="39"
           height="32"
           decoding="async"
           data-nimg="1"
           class="sc-iAlELC fNPfQt"
           srcset="/static/media/entities/assets/am.svg?w=48&amp;q=75 1x, /static/media/entities/assets/am.svg?w=96&amp;q=75 2x"
           src="/static/media/entities/assets/am.svg?w=96&amp;q=75"
           style="color: transparent;"
      >
    </a>
    <div class="sc-oeqTF hHgCDm">
      <a target="_self"
         data-testid="Link"
         class="sc-eEPDDI gughtU sc-ezTrPE bcttwp"
         href="/"
      >
        Home
      </a>
      <a target="_self"
         data-testid="Link"
         class="sc-eEPDDI gughtU sc-ezTrPE fZeUfM"
         href="/portfolio"
      >
        Portfolio
      </a>
      <a target="_self"
         data-testid="Link"
         class="sc-eEPDDI gughtU sc-ezTrPE fZeUfM"
         href="/about-me"
      >
        About
      </a>
      <a target="_self"
         data-testid="Link"
         class="sc-eEPDDI gughtU sc-ezTrPE fZeUfM"
         href="/blog"
      >
        Blog
      </a>
      <button data-testid="Button"
              type="button"
              class="sc-iCZwEW hJjhSG sc-drnuxz"
              style="display: none;"
      >
        <img alt="Notifications icon"
             loading="lazy"
             width="800"
             height="800"
             decoding="async"
             data-nimg="1"
             class="sc-hDcvty kKFXDJ"
             srcset="/static/media/entities/assets/notification.svg?w=828&amp;q=75 1x, /static/media/entities/assets/notification.svg?w=1920&amp;q=75 2x"
             src="/static/media/entities/assets/notification.svg?w=1920&amp;q=75"
             style="color: transparent; background: red; display: none;"
        >
      </button>
    </div>
  </nav>
</header>
`;
